tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -219,6 +220,13 @@ private async Task FetchConfigurationLoopAsync()
 
                     var api = requestFactory.Create(uri);
 
+                    // Add container ID header if available
+                    var containerId = ContainerMetadata.GetContainerId();
+                    if (containerId != null)
+                    {
+                        api.AddHeader(AgentHttpHeaderNames.ContainerId, containerId);
+                    }
```

Comment:
These are fixed for the lifetime of the app, right? In that case, we should likely include it as a default header instead, as part of the `CreateApiRequestFactory` call